moremesecons_mesechest/init.lua

```diff
@@ -104,8 +53,56 @@ default.chest.register_chest("moremesecons_mesechest:mesechest_locked", {
 	}
 })
 
-minetest.register_node = old_minetest_register_node
-minetest.register_lbm = old_minetest_register_lbm
+for _, chest in ipairs({"moremesecons_mesechest:mesechest", "moremesecons_mesechest:mesechest_locked",
+						"moremesecons_mesechest:mesechest_open", "moremesecons_mesechest:mesechest_locked_open"}) do
+	local old_def = minetest.registered_nodes[chest]
+
+	local old_on_metadata_inventory_put = old_def.on_metadata_inventory_put
+	local old_on_metadata_inventory_take = old_def.on_metadata_inventory_take
+	local old_on_rightclick = old_def.on_rightclick
+
+	local override = {}
+	override.on_metadata_inventory_put = function(pos, ...)
+		old_on_metadata_inventory_put(pos, ...)
+		mesecon.receptor_on(pos, {mesechest_get_output_rules(minetest.get_node(pos))[2]})
+		minetest.after(1, function(pos)
+			mesecon.receptor_off(pos, {mesechest_get_output_rules(minetest.get_node(pos))[2]})
+		end, pos)
+	end
+	override.on_metadata_inventory_take = function(pos, ...)
+		old_on_metadata_inventory_take(pos, ...)
+		mesecon.receptor_on(pos, {mesechest_get_output_rules(minetest.get_node(pos))[3]})
+		minetest.after(1, function(pos)
+			mesecon.receptor_off(pos, {mesechest_get_output_rules(minetest.get_node(pos))[3]})
+		end, pos)
+	end
+	override.on_rightclick = function(pos, node, clicker, ...)
+		if old_on_rightclick(pos, node, clicker, ...) == nil then
+			mesecon.receptor_on(pos, {mesechest_get_output_rules(node)[1]})
+		end
+	end
+
+	minetest.override_item(chest, override)
+end
+
+-- if the chest is getting closed, turn the signal off
+-- luacheck: ignore 122
+local old_lid_close = default.chest.chest_lid_close
+function default.chest.chest_lid_close(pn)
+	local pos = default.chest.open_chests[pn].pos
+	-- old_lid_close will return true if the chest won't be closed
+	if old_lid_close(pn) then
+		return true
+	else
+		local node = minetest.get_node(pos)
+		if node.name == "moremesecons_mesechest:mesechest" or
+				node.name == "moremesecons_mesechest:mesechest_open" or
+				node.name == "moremesecons_mesechest:mesechest_locked" or
+				node.name == "moremesecons_mesechest:mesechest_locked_open"then
```

Comment:
There's an alternative, possibly neater solution:
```Lua
local moremesecons_chests = {}
-- In the registration/overwrite loop:
moremesecons_chests[chest] = true
-- Here:
if moremesecons_chests[node.name] then
	mesecon.receptor_off(pos, {mesechest_get_output_rules(node)[1]})
end
```

Not a change request. Just an idea for you to consider.